workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Partials/Improbable.Coordinates

```diff
@@ -0,0 +1,68 @@
+public static readonly Coordinates Zero = new Coordinates(0, 0, 0);
+
+public static Coordinates operator -(Coordinates v1, Coordinates v2)
+{
+    return new Coordinates(v1.X - v2.X, v1.Y - v2.Y, v1.Z - v2.Z);
+}
+
+public static Coordinates operator +(Coordinates a, Coordinates b)
+{
+    return new Coordinates(a.X + b.X, a.Y + b.Y, a.Z + b.Z);
+}
+
+public static Coordinates operator *(Coordinates coords, float f)
+{
+    return new Coordinates(coords.X * f, coords.Y * f, coords.Z * f);
+}
+
+public static Coordinates operator *(Coordinates coords, double d)
+{
+    return new Coordinates(coords.X * d, coords.Y * d, coords.Z * d);
+}
+
+public static Coordinates operator /(Coordinates coords, float f)
+{
+    return new Coordinates(coords.X / f, coords.Y / f, coords.Z / f);
+}
+
+public static Coordinates operator /(Coordinates coords, double d)
+{
+    return new Coordinates(coords.X / d, coords.Y / d, coords.Z / d);
+}
+
+public static bool operator ==(Coordinates a, Coordinates b) => a.Equals(b);
+public static bool operator !=(Coordinates a, Coordinates b) => !a.Equals(b);
+
+public override bool Equals(object obj) 
```

Comment:
This is the only one we have defined Equals for. Even if not doing IEquatable in this pass I think we should still have this for the other ones too. Also I have a mild preference for having object overrides together, but not a strong one.

Reply:
Added `GetHashCode` and the equality operators to all the types.